packages/rs-drive-abci/src/execution/validation/state_transition/state_transitions/batch/action_validation/token/token_freeze_transition_action/state_v0/mod.rs

```diff
@@ -64,6 +72,30 @@ impl TokenFreezeTransitionActionStateValidationV0 for TokenFreezeTransitionActio
             return Ok(validation_result);
         }
 
+        // Check if the identity is already frozen
+        let (info, fee_result) = platform.drive.fetch_identity_token_info_with_costs(
```

Comment:
Don't we need to do it for all transition types that are affecting identity balance?